File: jacksandra-core/src/main/scala/com/mridang/jacksandra/CassandraStringSchema.scala
package com.mridang.jacksandra

import com.datastax.oss.driver.api.core.`type`.DataType
import com.fasterxml.jackson.databind.JavaType
import com.fasterxml.jackson.module.jsonSchema.types.StringSchema

/**
 * Wraps the string-schema returned by Jackson and contains the deduced corresponding
 * string type
 * <p>
 * https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cql_data_types_c.html
 *
 * @author mridang
 */
class CassandraStringSchema(val javaType: JavaType, val dataType: DataType)
  extends StringSchema with CassandraType {

  override def getDataType: DataType = dataType

  override def getJavaType: JavaType = javaType
}

File: jacksandra-core/src/main/scala/com/mridang/jacksandra/javabeans/CassandraStringSchema.java
package com.mridang.jacksandra.javabeans;

import com.datastax.oss.driver.api.core.type.DataType;
import com.fasterxml.jackson.databind.JavaType;
import com.fasterxml.jackson.module.jsonSchema.types.StringSchema;

/**
 * Wraps the string-schema returned by Jackson and contains the deduced corresponding
 * string type
 * <p>
 * https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cql_data_types_c.html
 *
 * @author mridang
 */
public class CassandraStringSchema extends StringSchema implements CassandraSchema {

    private final JavaType javaType;
    private final DataType dataType;

    public CassandraStringSchema(JavaType javaType, DataType dataType) {
        this.javaType = javaType;
        this.dataType = dataType;
    }

    @Override
    public DataType getDataType() {
        return dataType;
    }

    @Override
    public JavaType getJavaType() {
        return javaType;
    }
}
